spark/v3.1/spark/src/main/java/com/netease/arctic/spark/table/ArcticSparkTable.java

```diff
@@ -182,7 +194,17 @@ public ScanBuilder newScanBuilder(CaseInsensitiveStringMap options) {
 
   @Override
   public WriteBuilder newWriteBuilder(LogicalWriteInfo info) {
-    return new ArcticSparkWriteBuilder(arcticTable, info);
+    TableBlockerManager tableBlockerManager = catalog.getTableBlockerManager(arcticTable.id());
+    ArrayList<BlockableOperation> operations = Lists.newArrayList();
+    operations.add(BlockableOperation.BATCH_WRITE);
+    operations.add(BlockableOperation.OPTIMIZE);
+    Blocker block;
+    try {
+      block = tableBlockerManager.block(operations);
+    } catch (OperationConflictException e) {
+      throw new IllegalStateException("failed to block table " + arcticTable.id() + " with " + operations, e);
+    }
+    return new ArcticSparkWriteBuilder(arcticTable, info, tableBlockerManager, block);
```

Comment:
newWriteBuilder may be called during plan, request a block at buildForBatch() method, or even createBatchWriteFactory method